services/web/server/src/simcore_service_webserver/application_settings.py

```diff
@@ -1,91 +1,457 @@
+    )
+    WEBSERVER_STORAGE: Optional[StorageSettings] = Field(
+        auto_default_from_env=True, description="storage service client's plugin"
+    )
+    WEBSERVER_TRACING: Optional[TracingSettings] = Field(
+        auto_default_from_env=True, description="tracing plugin"
+    )
+
+    class Config(BaseCustomSettings.Config):
+        # NOTE: FutureWarning: aliases are no longer used by BaseSettings to define which environment variables to read.
+        #       Instead use the "env" field setting. See https://pydantic-docs.helpmanual.io/usage/settings/#environment-variable-names
+        # NOTE: These alias are ONLY used in export, not in constructor
+        fields = {
+            "SC_VCS_URL": "vcs_url",
+            "SC_VCS_REF": "vcs_ref",
+            "SC_BUILD_DATE": "build_date",
+            "SWARM_STACK_NAME": "stack_name",
+        }
+        alias_generator = lambda s: s.lower()
 
-    class Config:
-        env_prefix = "WEBSERVER_"
-        case_sensitive = False
-        alias_generator = snake_to_camel
+    # HELPERS  --------------------------------------------------------
 
-    # ---
+    def is_enabled(self, plugin_name: str):
+        return getattr(self, f"WEBSERVER_{plugin_name.upper()}", None) is not None
 
-    def public_dict(self) -> Dict:
+    def public_dict(self) -> Dict[str, Any]:
         """Data publicaly available"""
         return self.dict(
-            include={"app_name", "api_version", "vcs_url", "vcs_ref", "build_date"},
+            include={
+                "APP_NAME",
+                "API_VERSION",
+                "SC_VCS_URL",
+                "SC_VCS_REF",
+                "SC_BUILD_DATE",
+            },
             exclude_none=True,
+            by_alias=True,
         )
 
-    def to_client_statics(self) -> Dict:
-        return self.dict(
+    def to_client_statics(self) -> Dict[str, Any]:
+        data = self.dict(
             include={
-                "app_name",
-                "api_version",
-                "vcs_url",
-                "vcs_ref",
-                "build_date",
-                "swarm_stack_name",
+                "APP_NAME",
+                "API_VERSION",
+                "SC_VCS_URL",
+                "SC_VCS_REF",
+                "SC_BUILD_DATE",
+                "SWARM_STACK_NAME",
             },
             exclude_none=True,
             by_alias=True,
         )
+        # Alias in addition MUST be camelcase here
+        return {snake_to_camel(k): v for k, v in data.items()}
+
+
+def setup_settings(app: web.Application) -> ApplicationSettings:
+    app[APP_SETTINGS_KEY] = settings = ApplicationSettings.create_from_envs()
+    log.info("Captured app settings:\n%s", app[APP_SETTINGS_KEY].json(indent=1))
+    return settings
+
+
+def get_settings(app: web.Application) -> ApplicationSettings:
+    return app[APP_SETTINGS_KEY]
+
+
+def convert_to_app_config(app_settings: ApplicationSettings) -> Dict[str, Any]:
```

Comment:
so from now on and until trafaret is dropped, this is where the config must be changed? do I get that correctly?

Reply:
Changes to the trafaret config still needs to be done in the ``application_schema.py``. 
These two functions are here to map the trafaret config's "info" with the settings.
So, it is a bit more work to add a new section in the trafaret config ... but hopefully we drop it before this happens